library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/mapper/StringObfuscator.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.recorder.mapper
+
+internal class StringObfuscator {
+    fun obfuscate(stringValue: String): String {
+        return String(
+            CharArray(stringValue.length) {
+                val character = stringValue[it]
+                // Using the Character.isWhitespace() function we will support also the UNICODE
+                // characters according with the docs. Also have in mind that in JAVA the
+                // `char` primitive is a 16bits long number which covers all the UNICODE characters:
```

Comment:
`char` doesn't cover all the Unicode characters. It covers [BMP plain](https://en.wikipedia.org/wiki/Plane_(Unicode)) Unicode characters, but it doesn't cover Supplementary plain characters (above `0xFFFF`). When we call `character.code` we just cast from `char` to `int`, but to get `code` of the symbol from supplementary plain we need to merge 2 chars into single int.